crates/sui-graphql-rpc/src/extensions/logger.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::{
+    extensions::{
+        Extension, ExtensionContext, ExtensionFactory, NextExecute, NextParseQuery, NextRequest,
+        NextValidation,
+    },
+    parser::types::{ExecutableDocument, OperationType, Selection},
+    PathSegment, Response, ServerError, ServerResult, ValidationResult, Variables,
+};
+use std::{fmt::Write, sync::Arc};
+use tokio::sync::Mutex;
+use tracing::{error, info};
+use uuid::Uuid;
+
+// TODO: mode in-depth logging to debug
+
+#[derive(Clone, Debug)]
+pub struct LoggerConfig {
+    pub log_request_query: bool,
+    pub log_response: bool,
+    pub log_complexity: bool,
+}
+
+impl Default for LoggerConfig {
+    fn default() -> Self {
+        Self {
+            log_request_query: true,
+            log_response: true,
+            log_complexity: true,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct Logger {
+    config: LoggerConfig,
+}
+
+impl ExtensionFactory for Logger {
+    fn create(&self) -> Arc<dyn Extension> {
+        Arc::new(LoggerExtension {
+            session_id: None.into(),
+            config: self.config.clone(),
+        })
+    }
+}
+
+struct LoggerExtension {
+    session_id: Mutex<Option<Uuid>>,
+    config: LoggerConfig,
+}
+
+impl LoggerExtension {
+    async fn session_id(&self) -> Option<Uuid> {
+        *self.session_id.lock().await
+    }
+}
+
+#[async_trait::async_trait]
+impl Extension for LoggerExtension {
+    async fn request(&self, ctx: &ExtensionContext<'_>, next: NextRequest<'_>) -> Response {
+        *self.session_id.lock().await = Some(Uuid::new_v4());
+        next.run(ctx).await
+    }
+
+    async fn parse_query(
+        &self,
+        ctx: &ExtensionContext<'_>,
+        query: &str,
+        variables: &Variables,
+        next: NextParseQuery<'_>,
+    ) -> ServerResult<ExecutableDocument> {
+        let document = next.run(ctx, query, variables).await?;
+        let is_schema = document
```

Comment:
This seems maybe too broad? i.e. if the query happens to mention `__schema` then we won't log it, but what about if it's not the GraphQL `__schema`, i.e. it's on some other table, or if we are querying for the schema *and* something else?

Also, the length of L80 makes me think `cargo fmt` is not being run on this crate, or during CI?